src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpRequestMessageConfiguration.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Common;
+
+namespace NuGet.Protocol
+{
+    public class HttpRequestMessageConfiguration
```

Comment:
I chose not to, since most of the `*Context` types we have are mutable (and this is a similar concept).